### DISCUSSION TITLE
[Feature] Sort by Modified/Uploaded date in addition to the metadata defined date(created).

### DISCUSSION THREAD

**Discussion by LuminarLeaf** [immich-app/immich#8128] [Feature Request]
### The feature

Since we have gotten sorting support in the latest update anyway, maybe we can expand on it and provide sorting based on most(if not all) of the different dates stored for each asset in the database.
Currently stored fields:
![image](https://github.com/immich-app/immich/assets/80571430/a538bc2c-f17d-4039-a3c6-1ae095273ea7)

Also if this is implemented then the sort direction can be another toggle or the current one can remain and we can have a new one named something like sorting mode which will add the aforementioned types.

This will be a great boon for people like me who use immich to have a pseudo stash of wallpapers and want it sorted by lets say upload time to view what we uploaded recently instead of uploading something with the metadata date of 2006 and then losing it in the enormous amounts of other stuff. (In case you are wondering, yes I have hosted my own instance for this exact purpose and I'm not going to share the link because it is hosted on my laptop and only me and some of my friend have the access.)

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by glowtape** [comment]
Would like to have this, too. This would making ingress of large amounts of images easier, especially when retroactively tagging them and putting them in albums.

**Comment by thevrownguy** [comment]
Please please add this. When importing pictures that might be from a different year, or may be tagged with the wrong year, this is a must.

**Comment by Delivator** [comment]
Yes pleaseee

**Comment by deWasbeer** [comment]
agree :)

**Comment by bverkron** [comment]
This feature would also be great for multi-user setups. I am slowly making my way through a very large back catalog of family photos in Lightroom and periodically add much older photos (5, 10 or even 15 years old) into Immich but my family members currently have no way to know when they are added or easily find them. Being able to sort by date added would allow them to easily see "new" photos I've added to Immich, even if they are taken a long time ago.

**Comment by Qhilm** [comment]
Another use case: I share public links after events and ask everyone to upload. It's a mess at the moment to understand who uploaded what and when. I currently resort to sorting by camera model, which only works when they do not have the same smartphone model as me for example.

Additionally (probably a different FR though), a sort of mini timeline inside the "comments" section, to see who uploaded what and when, would be great.

**Comment by PangeranWiguan** [comment]
Yes, I vote to add this feature as well.

**Comment by stefan1983** [comment]
A feature to show "last batch / manual import" would be nice!

+1

**Comment by Shyryp** [comment]
This feature is sorely lacking in Immich. Currently, it's impossible to find out what was recently loaded into the Immich library (via the web interface or via immich-go).

**Reply by tobse91** [reply]
I always create a new album when using immich-cli as a workaround. Then I know which pictures I recently uploaded. 
But via webui this isn't possible and sometimes I can't find my new uploaded pictures.

**Reply by Shyryp** [reply]
> I always create a new album when using immich-cli as a workaround. Then I know which pictures I recently uploaded. But via webui this isn't possible and sometimes I can't find my new uploaded pictures.

Yes, that's good advice 👍

But for me, for example, photos are also uploaded directly from my smartphone (using Immich for Android) when my smartphone is on the same network as my PC (the Immich server). This happens essentially automatically and continuously (so as not to have to do it manually every few months), which means I lose the ability to track when new photos were uploaded (the photo creation date alone won't help sorting in this case, since my smartphone sometimes receives photos from other devices with older creation dates).

So I still hope the Immich developers will add metadata indicating the upload date of photos and videos to the database.

**Reply by skatsubo** [reply]
@Shyryp 
> add metadata indicating the upload date of photos and videos to the database

The field is already there: `asset.createdAt` in Postgres.
Even more, it is returned by some API calls: https://api.immich.app/models/AssetResponseDto
```
createdAt
DateTime
The UTC timestamp when the asset was originally uploaded to Immich.
```

**If I understand it correctly, the problem is in exposing it through API in a "usable" way.**

Currently:
1. The timeline implementation uses the Buckets API https://api.immich.app/endpoints/timeline which is centered around original datetime of assets.
2. Same for the Search API https://api.immich.app/endpoints/search/searchAssets. It paginates and sorts results by original datetime `fileCreatedAt`. See example:
<details><summary>API call with curl</summary>

```sh
curl -sSL -X POST -H 'Cookie: immich_access_token=JIeHyWr9gWDEnXJn6OUJ0zl1UKHgxbT4bXh8Z7Xo' 'https://my.immich.app/api/search/metadata' | jq -r '.assets.items[] | "\(.originalPath), createdAt: \(.createdAt), fileCreatedAt: \(.fileCreatedAt)"' | head -100 | tail
/photo/sk/2025/2025-09/IMG_20250913_125533.jpg, createdAt: 2025-09-13T12:00:04.136Z, fileCreatedAt: 2025-09-13T09:55:34.149Z
/photo/sk/2025/2025-09/IMG_20250913_125522.jpg, createdAt: 2025-09-13T12:00:04.136Z, fileCreatedAt: 2025-09-13T09:55:23.296Z
/photo/sk/2025/2025-09/IMG_20250913_125356.jpg, createdAt: 2025-09-13T12:00:04.136Z, fileCreatedAt: 2025-09-13T09:53:58.730Z
/photo/kk/2025/IMG_20250909_120243.jpg,         createdAt: 2025-11-03T07:00:02.268Z, fileCreatedAt: 2025-09-09T12:02:45.310Z
/photo/kk/2025/IMG_20250909_120238.jpg,         createdAt: 2025-11-03T07:00:02.268Z, fileCreatedAt: 2025-09-09T12:02:40.970Z
/photo/kk/2025/IMG_20250909_120234.jpg,         createdAt: 2025-11-03T07:00:02.268Z, fileCreatedAt: 2025-09-09T12:02:37.580Z
/photo/kk/2025/IMG_20250909_120220.jpg,         createdAt: 2025-11-03T07:00:02.268Z, fileCreatedAt: 2025-09-09T12:02:24.340Z
/photo/kk/2025/IMG_20250909_120216.jpg,         createdAt: 2025-11-03T07:00:02.268Z, fileCreatedAt: 2025-09-09T12:02:19.520Z
/photo/kk/2025/IMG_20250909_120138.jpg,         createdAt: 2025-11-03T07:00:02.268Z, fileCreatedAt: 2025-09-09T12:01:40.500Z
/photo/kk2/2025/20250903_082343.JPG,            createdAt: 2025-09-03T12:00:03.822Z, fileCreatedAt: 2025-09-03T05:23:43.942Z
```
</details>

**Comment by janstadt** [comment]
Any update on the progress here? The wife was just complaining about this. Seems like a no brainer.

**Comment by mrpatricko** [comment]


**Reply by bo0tzz** [reply]
Please don't comment to vote, but just use the actual vote button.

**Comment by photoserver-pro** [comment]
I would like to strongly support this feature request and add an important extension to it.

Sorting by Modified date / Uploaded date in addition to the metadata-defined creation date is essential, especially for users who ingest large existing libraries or regularly add historical material.

From my perspective, this sorting option should be globally consistent across all views, not limited to the main timeline only. Concretely, the same sort modes should be available in:
	•	Tags
	•	People
	•	Places / Locations
	•	Albums (including shared albums)
	•	Folders / External Libraries

Without this, navigation becomes inconsistent: users may see a different chronological order depending on where they are browsing, even though they are looking at the same assets.

For large installations (hundreds of thousands of assets), being able to switch between:
	•	metadata-based creation date
	•	file modified date
	•	upload / ingest date

in every view is critical for usability and discoverability. This is especially true when working with mixed archives, rescans, or incremental imports via external libraries.

In short:
Sorting should be treated as a global browsing concern, not a view-specific feature.

Thanks for considering this — it would significantly improve day-to-day navigation for power users and large multi-library setups.

**Comment by Erosion139** [comment]
Please add this I just mistakenly had a mass upload to the wrong account and now my main account has random images from someone else. Cant believe we dont have sort by date added yet have all these special context search features.

**Comment by Vogeluff** [comment]
i would even go so far and say it would be a nice alternative or addition for the top space in the timeline where you can currently have memories. memories are great but I think I'd even prefer a little "last added" hightlightreel or something that perhaps could be configurable for last X days/weeks or so.

**Reply by Erosion139** [reply]
I was just told that a recently uploaded view is coming in V3
<img width="531" height="139" alt="image" src="https://github.com/user-attachments/assets/a810c920-2956-427e-a6cf-d87afd023ae3" />

**Comment by LuminarLeaf** [comment]
The new recently uploaded view is great but thats not what the feature request was for.

The feature request is simply to give more sorting options in albums (and maybe the timeline).

**Reply by bo0tzz** [reply]
This feature request is still open

**Reply by LuminarLeaf** [reply]
Oops, should've replied to the comment instead of the thread, my bad. ;-;
